workflows/templates/wipe-and-reboot-worker.yaml

```diff
@@ -39,7 +39,41 @@ spec:
           - name: bootTimeoutInSeconds
       dag:
         tasks:
+        - name: wait-for-keycloak
+          templateRef:
+            name: kubectl-and-curl-template
+            template: shell-script
+          arguments:
+            parameters:
+              - name: dryRun
+                value: "{{inputs.parameters.dryRun}}"
+              - name: scriptContent
+                value: |
+                  client_secret=$(kubectl get secrets admin-client-auth -o jsonpath='{.data.client-secret}' | base64 -d)
+                  echo "Sleeping for 60 seconds to let keycloak settle after node drain"
+                  sleep 60
+                  count=0
+                  total=120
+                  sleep=5
+                  while true; do
+                    http_code=$(curl -k -s -S -o /dev/null -w '%{http_code}' -d grant_type=client_credentials -d client_id=admin-client \
+                        -d client_secret="$client_secret" https://api-gw-service-nmn.local/keycloak/realms/shasta/protocol/openid-connect/token || true)
+                    if [ "${http_code}" == "200" ]; then
+                      echo "keycloak is up"
+                      break
+                    else
+                      if [ "${count}" == "${total}" ]; then
+                          echo "keycloak is down after ${total} checks, giving up ..."
+                          exit 1
+                      fi
+                      count=$(($count + 1))
+                      echo "keycloak is down, sleeping for $sleep seconds and retry, attempt $count/$total ..."
+                      sleep $sleep
+                    fi
+                  done
         - name: "validate-bss-ntp"
+          dependencies:
+          - wait-for-keycloak
```

Comment:
Added wait condition for keycloak pod, which may have been migrated at previous step and did not yet start.